src/typeInference/annotateFlworExpression.ts

```diff
@@ -178,11 +182,15 @@ function annotateOrderByClause(ast: IAST, annotationContext: AnnotationContext):
  * @param array the array to be filtered
  * @returns the filtered array
  */
-function filterOnUniqueObjects(array: SequenceType[]): SequenceType[] {
+export function filterOnUniqueObjects(array: SequenceType[]): SequenceType[] {
 	return array.filter(
 		(current, index, arrayCopy) =>
 			arrayCopy.findIndex(
-				(element) => element.type === current.type && element.mult === current.mult
+				(element) =>
+					element &&
```

Comment:
Why are these two checks for element / current needed? If the input array contains nulls, I'd prefer to address that where it is being done. This might swipe some mistakes under the rug.

Reply:
Null checks are moved to annotateForClause